### PR TITLE
More Mesh Updates

### PR DESCRIPTION
**Pull request by bryab**
Some name updates from looking at the "Skin Export" text files that are present in some S3D files.

I was hoping this would make the "data9" field make more sense - as to why the "type" is at the end - but it didn't.  I have at least renamed this field to "meshops" and wrote some notes about the name of each meshop.

### CODE REVIEW

File: crates/libeq_wld/src/parser/fragments/mesh.rs [hunk 1]
```diff
@@ -155,37 +156,39 @@ pub struct MeshFragment {
...
     /// references.
-    pub vertex_materials: Vec<(u16, u16)>,
+    pub vertex_material_groups: Vec<(u16, u16)>,
 
-    /// _Unknown_ - A collection of [MeshFragmentData9Entry]s
-    pub data9: Vec<MeshFragmentData9Entry>,
+    /// _Unknown_ - A collection of [MeshFragmentMeshOpEntry]s
+    pub meshops: Vec<MeshFragmentMeshOpEntry>,
+
+    // FIXME: There's some trailing bytes here, between 1-3 in _chr wlds.
```
Comment (cjab) [inline crates/libeq_wld/src/parser/fragments/mesh.rs]: Is this what's causing the new test to fail?
It sounds a lot like padding to make sure this falls on a 4-byte word boundary. Which I _think_ makes sense here. It looks like depending on the `type_field` in the mesh op entries you could end up with a partial word at the end?

Comment (bryab) [reply crates/libeq_wld/src/parser/fragments/mesh.rs]: I'll look into that!  Yes, I made that new test to illustrate this problem.  Is there another fragment with this issue that you resolved?

Comment (cjab) [reply crates/libeq_wld/src/parser/fragments/mesh.rs]: Yep! Check out `user_data` on `RegionFlagFragment`.

The `into_bytes` function on `StringHash` also has to _add_ padding.

Comment (bryab) [reply crates/libeq_wld/src/parser/fragments/mesh.rs]: Just to confirm: I am not actually reading the zeroes when parsing, but I am adding them when serializing.

Comment (cjab) [reply crates/libeq_wld/src/parser/fragments/mesh.rs]: Yep!

**Comment by cjab** [comment]
There were a few small `use` conflicts but this has been rebased and merged!

**Pull request closed without merging**